src/main/java/fi/oph/akt/service/email/ClerkEmailService.java

```diff
@@ -31,18 +54,66 @@ public void createInformalEmails(InformalEmailRequestDTO emailRequestDTO) {
 		}
 
 		translators.forEach(translator -> {
-			// @formatter:off
-            // TODO: replace recipient with translator's email address
-            EmailData emailData = EmailData.builder()
-                    .sender("AKT")
-                    .recipient("translator" + translator.getId() + "@test.fi")
-                    .subject(emailRequestDTO.subject())
-                    .body(emailRequestDTO.body())
-                    .build();
-            // @formatter:on
-
-			emailService.saveEmail(EmailType.INFORMAL, emailData);
+			// TODO: replace recipient with translator's email address
+			createEmail("translator" + translator.getId() + "@test.fi", emailRequestDTO.subject(),
+					emailRequestDTO.body(), EmailType.INFORMAL);
 		});
 	}
 
+	private Email createEmail(final String recipient, final String subject, final String body,
+			final EmailType emailType) {
+
+		// @formatter:off
+		final EmailData emailData = EmailData.builder()
+				.sender("AKT")
+				.recipient(recipient)
+				.subject(subject)
+				.body(body)
+				.build();
+		// @formatter:on
+
+		return emailService.saveEmail(emailType, emailData);
+	}
+
+	@Transactional
+	public void createAuthorisationExpiryEmail(final AuthorisationTerm authorisationTerm) {
```

Comment:
Jostakin syystä ei tunnu hyvältä että entiteettiä passataan yhdestä transaktiosta ulos ja toiseen transaktioon sisään. Testien mukaanhan se toimii. Mutta, se toimii vain koska detachattu entiteettillä on jo se data mitä täällä tarvitaan. Onko näin aina, en ole varma.

Reply:
Viittaatko EmailServicen `saveEmail` metodin palauttamaan Email-entiteettiin vai tähän AuthorisationTerm:iin? Vai kenties molempiin?

Reply:
Tarkoitin authorisationTerm:iä, antaisin id:n sisään kun mennään transaktiosta toiseen. Ja kyllä tuo email servicen kutsukin on vähän ongelmallinen, kun transaktioraja on hieman häilyvä, mutta se on eri asia eikä rajoitu vain tähän luokkaan.